### DISCUSSION TITLE
Improvement for Protein Usage Calculation in reportEnzymeUsage

### DISCUSSION THREAD

**Discussion by HossFir** [SysBioChalmers/GECKO#387] [Ideas]
In the `reportEnzymeUsage` function, I suggest an improvement for calculating protein usage. Currently, the method can result in some proteins showing unrealistically high usage percentages, especially for proteins not reported in the proteomics data and having a lower bound of -1000.

To address this, I propose setting the protein usage to the minimum calculated value from the previous steps (STEP 69-70). Additionally, the protein pool should be adjusted to match this minimum protein usage. This change would help prevent very high usage percentages, such as 80-90%, especially for proteins that are not present in the proteomics data, leading to more accurate and realistic protein usage calculations.

**Comment by edkerk** [comment]
I'm not sure I'm following your thoughts here, could you please be a little more specific? `reportEnzymeUsage` gives two sets of enzyme usage data, `highUsage` and `topUsage`, which (arguably somewhat confusingly) have the fields 'highUsage.absUsage`; `highUsage.capUsage`; `topUsage.absUsage` and highUsage.percUsage`. The two `absUsage` are direct absolute flux values, but `capUsage` is calculated by dividing `absUsage` by the LB of each enzyme usage reaction (which is -1000 without proteomics data); while `percUsage` is calculated by dividing `absUsage` by the LB of the protein pool exchange reaction (which since [GECKO 3.2.0](https://github.com/SysBioChalmers/GECKO/issues/375) does not change when proteomics data is integrated, unless explicitly done so using `setProtPoolSize`).

Regardless, I fail to see how either `capUsage` or `percUsage` for proteins without proteomics data would have particularly high protein values?

At the same time, I agree with the suggestion that `percUsage` should be calculated from the protein pool exchange __flux__, and not the protein pool exchange __lower bounds__.

**Reply by HossFir** [reply]
Sorry for not explaining thoroughly!

Here’s what I mean: When optimizing the model for minimal protein usage, as shown in the script and protocol below:

```
ecModel = loadEcModel('ecModel.yml');
ecModel = setParam(ecModel, 'obj', params.bioRxn, 1);
sol = solveLP(ecModel);
fprintf('Growth rate that is reached: %f /hour.\n', sol.f);
ecModel = setParam(ecModel, 'lb', params.bioRxn, 0.99 * sol.f);
ecModel = setParam(ecModel, 'obj', 'prot_pool_exchange', 1);
sol = solveLP(ecModel);
fprintf('Minimum protein pool usage: %.2f mg/gDCW.\n', sol.f);
```

Instead of minimizing glucose exchange and setting a lower bound for biomass reactions, we can use the fluxes obtained from the minimum protein usage optimization:

```
ecModel = setParam(ecModel, 'obj', 'EX_glc__D_e', 1);
ecModel = setParam(ecModel, 'lb', params.bioRxn, 0.42493);
sol = solveLP(ecModel, 1);
```

However this method is somewhat arbitrary, using fluxes from the minimum total protein usage optimization might be somehow useful, meaning that the proteins that were not reported in the proteomics dataset and had the `LB` of `-1000` will likely have lower protein usage as the optimization aimed at minimizing total protein usage. In this case, we can replace:

```
protPool = -ecModel.lb(strcmp(ecModel.rxns, 'prot_pool_exchange'));
```

with the calculated minimum protein usage.

This gives us enzyme usage based on minimizing total protein usage, with the percentage representing the ratio of protein usage to the minimized protein usage. However, this is just an idea, and the protocol provided in the GECKO pipeline is perfectly fine!
